crema/parsers/crux.py

```diff
@@ -64,55 +69,58 @@ def read_crux(txt_files, peptide_tdc=False, copy_data=True):
     if not scores:
         raise ValueError(
             "Could not find any of the Crux score columns in all of the files."
-            f"The columns crema looks for are {', '.join(list(scores))}"
+            f" The columns Crema looks for are {', '.join(list(scores_all))}"
         )
 
     scores = list(scores)
 
     # Read in the files:
-    fields = spectrum + [peptide] + [target] + scores + [pairing]
+    fields = spectrum + [peptide] + [target] + scores + [pairing] + [protein]
     if isinstance(txt_files, pd.DataFrame):
         data = txt_files.copy(deep=copy_data).loc[:, fields]
     else:
-        data = pd.concat([_parse_psms(f, fields) for f in txt_files])
+        data = pd.concat(
+            [utils.parse_psms_txt(f, fields, False) for f in txt_files]
+        )
 
     psms = read_txt(
         data,
         target_column=target,
         spectrum_columns=spectrum,
         score_columns=scores,
         peptide_column=peptide,
+        protein_column=protein,
+        protein_delim=protein_delim,
         sep="\t",
         copy_data=False,
     )
-    if peptide_tdc:
-        psms._peptide_pairing = _create_pairing(data)
-
-    return psms
 
+    # always pair target and decoys for Crux
+    if pairing_file_name == None:  # implicit pairing
+        psms._peptide_pairing = _create_pairing(data)
+    else:  # explicit pairing
+        psms._peptide_pairing = utils.create_pairing_from_file(
+            pairing_file_name
+        )
 
-def _parse_psms(txt_file, cols, log=True):
-    """Parse a single Crux tab-delimited file
-
-    Parameters
-    ----------
-    txt_file : str
-        The crux tab-delimited file to read.
-    cols : list of str
-        The columns to parse.
+    # Remove the start position of peptide in protein if present
+    # This looks like "protName(XX)"
+    # Remove decoy prefix from protein ID
+    protein_column = psms.proteins
+    new_protein_column = protein_column.str.replace(
+        "\\([^()]*\\)", "", regex=True
+    )
+    new_protein_column = new_protein_column.str.replace(
+        "decoy_", "", regex=True
```

Comment:
**question:** Isn't the decoy prefix a parameter for the search engine?

Reply:
Yes, good point. I was using the default here. I'll add an argument